includes/admin/js/geolocator.js

```diff
@@ -1,53 +1,194 @@
+		$( '#wanderlist-region' ).val( region );
+		$( '#wanderlist-country' ).val( country );
+		$( '#wanderlist-lng' ).val( lng );
+		$( '#wanderlist-lat' ).val( lat );
+		var placeName = feature.place_name;
+
+		// Since Geocoder results for Vatican aren't accurate, we'll overwrite them
+		if ( 'Città del Vaticano' === city ) {
+			$( '#wanderlist-region' ).val( '' );
+			$( '#wanderlist-country' ).val( 'Vatican City' );
+			placeName = 'Città del Vaticano';
+		}
+
+		// Show a friendly success message and show point on map
+		$( '#wanderlist-geocoder-message' ).removeClass( 'error' );
+		$( '#wanderlist-geocoder-message' ).addClass( 'success' );
+		$( '#wanderlist-geocoder-message' ).find( '.place' ).text( placeName );
+		showPointOnMap( lat, lng );
+	}
+
+	/*
+	 * Get the user's current location.
+	 * This uses the HTML5 geolocation API, available on
+ 	 * most mobile browsers and modern browsers
+	 * http://caniuse.com/#feat=geolocation
+	 */
+	 function getCurrentLocation( button ) {
+		 // If our browser doesn't support geolocation, hide the button for a more graceful degredation
+		 if ( ! navigator.geolocation ) {
+			 button.innerHTML = 'Geolocation is not available';
+			 $( button ).hide();
+		 // Otherwise, locate user when the button is clicked
+		 } else {
+			 button.onclick = function( e ) {
+				 e.preventDefault();
+			   e.stopPropagation();
+				 $( button ).siblings( '.wanderlist-loader' ).addClass( 'active' );
+			   map.locate();
+				};
 			}
-		})
-	});
+
+			// Once we've got a position, zoom and center the map on it, and add a single marker.
+			map.on( 'locationfound', function( e ) {
+				geocodeCoords( e.latlng.lat, e.latlng.lng );
+				map.fitBounds( e.bounds );
+				showPointOnMap( e.latlng.lat, e.latlng.lng );
+				$( button ).siblings( '.wanderlist-loader' ).removeClass( 'active' );
+			});
+
+			// If the user chooses not to allow their location to be shared, display an error message.
+			map.on( 'locationerror', function() {
+				$( '#wanderlist-geocoder-message' ).removeClass( 'success' );
+				$( '#wanderlist-geocoder-message' ).addClass( 'error' );
+				$( '#wanderlist-geocoder-message' ).find( '.error-message' ).text( 'Please share your location so we can find you!' );
+			});
+		}
+
+		/*
+     * This reverse geocodes a set of lat/long coordinates.
+     * Used when we've only been given the coordinates, and we
+		 * need some more information.
+     */
+		 function geocodeCoords( lat, lon ) {
+			 var geocodeURI = 'https://api.mapbox.com/geocoding/v5/mapbox.places/' + encodeURI( lon ) + ',' +  + encodeURI( lat ) + '.json?access_token=' + L.mapbox.accessToken;
+			 $.ajax( geocodeURI, {
+				success: function( response ) {
+					parseLocationData( response.features[0] );
+				},
+				error: function( response ) {
+					console.log( response )
+				}
+			})
+		}
+
+		/*
+ 		 * This shows the selected point on a map.
+ 	 	 *
+ 	 	 */
+		 function showPointOnMap( lat, lon ) {
```

Comment:
Longitude is usually shortened to `lng` in my dealings with geo code, so I'd change this from `lon` to `lng`.